src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceVariableService.State.cs

```diff
@@ -186,6 +186,11 @@ public SemanticMap GetSemanticMap(CancellationToken cancellationToken)
             private TExpressionSyntax GetExpressionUnderSpan(SyntaxTree tree, TextSpan textSpan, CancellationToken cancellationToken)
             {
                 var root = tree.GetRoot(cancellationToken);
+                if (textSpan.Length == 0)
+                {
+                    return root.FindToken(textSpan.Start).GetAncestor<TExpressionSyntax>();
```

Comment:
probably better to do "return root.FindToken(...).Parent as TExpressionSyntax;".  That way it works for tokens that are directly parented by expressions (like the => in a lambda), but not some random token (like ```if``` or ```switch```) in a lambda.